catalog_client/agents/node_agent.py

```diff
@@ -20,4 +22,6 @@ def collect(self):
             env=environ,
         ).addCallback(
             lambda version: version.strip()
+        ).addErrback(
+            lambda _ : "Unknown Version"
```

Comment:
The only way I now on coreOS is to call out to docker 

```
docker run clusterhq/flocker-dataset-agent flocker-diagnostics --version
```

IMO - not elegant but will work.

I think is Flocker becomes VH aware it can, if configured, report back to VH.